### PR TITLE
65293_user_search_team

### PR DESCRIPTION
**Pull request by yasin-buet**
sss

### CODE REVIEW

**Comment by asifbuetcse** [comment]
OK

File: app/src/Controller/TeamsController.php [hunk 1]
```diff
@@ -25,13 +26,23 @@ public function isAuthorized($user)
      */
     public function index()
     {
-        $this->paginate = [
-            'contain' => ['Countries', 'Continents']
-        ];
+        $this->paginate = ['contain' => ['Countries', 'Continents']];
         $teams = $this->paginate($this->Teams);
+        $this->loadModel('Countries');
```
Comment (tuandv2611) [inline app/src/Controller/TeamsController.php]: Should load it in construct function
In any action don't need defined as this

File: app/src/Controller/TeamsController.php [hunk 2]
```diff
@@ -25,13 +26,23 @@ public function isAuthorized($user)
...
+        $countries = $this->Countries->find('list', [
```
Comment (tuandv2611) [inline app/src/Controller/TeamsController.php]: countries don't use for view?

File: app/src/Controller/TeamsController.php [hunk 3]
```diff
@@ -25,13 +26,23 @@ public function isAuthorized($user)
...
+            $teams = TableRegistry::get('Teams');
+            $query = $teams->find();
+            $query->where(['Countries.name LIKE' => '%' . $this->request->data['keyword'] . '%']);
+            $teams = $this->paginate($query);
+        }
 
         $this->set(compact('teams'));
-        $this->set('_serialize', ['teams']);
+        $this->set(compact('countries'));
+        $this->set('_serialize', ['teams', 'countries']);
```
Comment (tuandv2611) [inline app/src/Controller/TeamsController.php]: why have to use _serialize?

File: app/src/Controller/TeamsController.php [hunk 4]
```diff
@@ -25,13 +26,23 @@ public function isAuthorized($user)
...
         $teams = $this->paginate($this->Teams);
+        $this->loadModel('Countries');
+        $countries = $this->Countries->find('list', [
+            'keyField' => 'id',
+            'valueField' => 'name'
+        ]);
+        if ($this->request->is('post')) {
+            $teams = TableRegistry::get('Teams');
+            $query = $teams->find();
+            $query->where(['Countries.name LIKE' => '%' . $this->request->data['keyword'] . '%']);
```
Comment (tuandv2611) [inline app/src/Controller/TeamsController.php]: search with only name of country? I think that it's not enough for require "Can search team by keyword such as: country, continent"

File: app/src/Template/Teams/index.ctp [hunk 5]
```diff
@@ -1,3 +1,15 @@
...
+            <?= $this->Form->create() ?>
+            <?= $this->Form->input('keyword', ['class' => 'form-control']) ?>
+            <?= $this->Form->button('Search', ['class' => 'btn btn-primary ']) ?>
+            <?= $this->Form->end() ?>
+        </div>
+    </div>
+</div>
 <section id="contentbody">
     <div class="container">
         <?php foreach ($teams as $teams): ?>
```
Comment (tuandv2611) [inline app/src/Template/Teams/index.ctp]: I know this code in last pull, but you need change it.

`<?php foreach ($teams as $team): ?>`

File: app/src/Template/Teams/index.ctp [hunk 6]
```diff
@@ -1,18 +1,32 @@
...
+                    <h2><?= h($team->name) ?></h2>
                 </div>
                 <div class="single_stuff_article">
                     <div class="single_sarticle_inner">
                         <div class="stuff_article_inner">
-                            <span class="stuff_date"><strong><?= h($teams->modified) ?> </strong>
-                                <h2><?= h($teams->title) ?></a></h2>
-                                    <p><?= h($teams->description) ?></p>
+                            <span class="stuff_date"><strong><?= h($team->modified) ?> </strong>
+                                <h2><?= h($team->title) ?></a></h2>
```
Comment (tuandv2611) [inline app/src/Template/Teams/index.ctp]: remove </a> 
and below

File: app/src/Template/Teams/index.ctp [hunk 7]
```diff
@@ -1,18 +1,32 @@
...
+                                <h3><?= __('Country Name : ') ?><?= h($team->country->name) ?></a></h3>
+                                <h3><?= __('Continent Name : ') ?><?= h($team->continent->name) ?></a></h3>
+                                    <p><?= h($team->description) ?></p>
```
Comment (tuandv2611) [inline app/src/Template/Teams/index.ctp]: indent

File: app/src/Template/Teams/index.ctp [hunk 8]
```diff
@@ -1,18 +1,32 @@
...
-                    <h2><?= h($teams->name) ?></h2>
+                    <h2><?= h($team->name) ?></h2>
                 </div>
                 <div class="single_stuff_article">
                     <div class="single_sarticle_inner">
                         <div class="stuff_article_inner">
-                            <span class="stuff_date"><strong><?= h($teams->modified) ?> </strong>
-                                <h2><?= h($teams->title) ?></a></h2>
-                                    <p><?= h($teams->description) ?></p>
+                            <span class="stuff_date"><strong><?= h($team->modified) ?></strong>
```
Comment (tuandv2611) [inline app/src/Template/Teams/index.ctp]: where close tag <span> ?
indent for all below tag

File: app/src/Controller/TeamsController.php [hunk 9]
```diff
@@ -25,13 +26,17 @@ public function isAuthorized($user)
      */
     public function index()
     {
-        $this->paginate = [
-            'contain' => ['Countries', 'Continents']
-        ];
+        $this->paginate = ['contain' => ['Countries', 'Continents']];
         $teams = $this->paginate($this->Teams);
+        if ($this->request->is('post')) {
+            $teams = TableRegistry::get('Teams');
```
Comment (TungPro) [inline app/src/Controller/TeamsController.php]: combine this line with bottom line

**Comment by tuandv2611** [comment]
OK

**Comment by TungPro** [comment]
OK

**Pull request merged**